nyxx_interactions/lib/src/builders/CommandPermissionBuilder.dart

```diff
@@ -0,0 +1,42 @@
+part of nyxx_interactions;
+
+/// Used to define permissions for a particular command.
+abstract class ICommandPermissionBuilder extends Builder {
+
+  /// The ID of the Role or User to give permissions too
+  final Snowflake id;
+
+  /// Does the role have permission to use the command
+  final bool hasPermission;
+
+  ICommandPermissionBuilder(this.id, {this.hasPermission = true});
+
```

Comment:
here you coud add factory constructor for subclasses -- that would be very convinient